src/hotspot/cpu/x86/stubGenerator_x86_64_aes.cpp

```diff
@@ -264,6 +291,92 @@ address StubGenerator::generate_galoisCounterMode_AESCrypt() {
   return start;
 }
 
+// AVX2 Vector AES Galois Counter Mode implementation.
+//
+// Inputs:           Windows    |   Linux
+//   in         = rcx (c_rarg0) | rsi (c_rarg0)
+//   len        = rdx (c_rarg1) | rdi (c_rarg1)
+//   ct         = r8  (c_rarg2) | rdx (c_rarg2)
+//   out        = r9  (c_rarg3) | rcx (c_rarg3)
+//   key        = rdi           | r8  (c_rarg4)
+//   state      = r13           | r9  (c_rarg5)
+//   subkeyHtbl = r11           | r11
+//   counter    = rsi           | r12
+//
+// Output:
+//   rax - number of processed bytes
+address StubGenerator::generate_avx2_galoisCounterMode_AESCrypt() {
+  __ align(CodeEntryAlignment);
+  StubCodeMark mark(this, "StubRoutines", "galoisCounterMode_AESCrypt");
+  address start = __ pc();
+
+  const Register in = c_rarg0;
+  const Register len = c_rarg1;
+  const Register ct = c_rarg2;
+  const Register out = c_rarg3;
+  // and updated with the incremented counter in the end
+ #ifndef _WIN64
+  const Register key = c_rarg4;
+  const Register state = c_rarg5;
+  const Address subkeyH_mem(rbp, 2 * wordSize);
+  const Register subkeyHtbl = r11;
+  const Address counter_mem(rbp, 3 * wordSize);
+  const Register counter = r12;
+ #else
+  const Address key_mem(rbp, 6 * wordSize);
+  const Register key = rdi;
+  const Address state_mem(rbp, 7 * wordSize);
+  const Register state = r13;
+  const Address subkeyH_mem(rbp, 8 * wordSize);
+  const Register subkeyHtbl = r11;
+  const Address counter_mem(rbp, 9 * wordSize);
+  const Register counter = rsi;
+ #endif
+  __ enter();
+  // Save state before entering routine
+  __ push(r12);
+  __ push(r13);
+  __ push(r14);
+  __ push(r15);
+  __ push(rbx);
+#ifdef _WIN64
+  // on win64, fill len_reg from stack position
+  __ push(rsi);
+  __ push(rdi);
+  __ movptr(key, key_mem);
+  __ movptr(state, state_mem);
+#endif
+  __ movptr(subkeyHtbl, subkeyH_mem);
+  __ movptr(counter, counter_mem);
+
+  // Save rbp and rsp
+  __ push(rbp);
+  __ movq(rbp, rsp);
+  // Align stack
+  __ andq(rsp, -64);
+  __ subptr(rsp, 16 * longSize); // Create space on the stack for saving AES entries
+
+  aesgcm_avx2(in, len, ct, out, key, state, subkeyHtbl, counter);
+  __ vzeroupper();
+  __ movq(rsp, rbp);
+  __ pop(rbp);
```

Comment:
If you remove `movq(rbp, rsp)` above, you can replace this with:
```suggestion
  __ lea(rsp, Address (rbp, WINDOWS_ONLY(-7) NOT_WINDOWS(-5) * wordSize));
```